examples/dyn_phy.py

```diff
@@ -19,104 +21,133 @@
 import gt4py
 from fv3gfs.physics.stencils.physics import Physics
 
-# Reuse infrastructure to read in grid variables
-# add path to integration test to reuse existing grid logic
-sys.path.append("/port_dev/tests/savepoint/translate/")
-from translate_update_dwind_phys import TranslateUpdateDWindsPhys
+import click
+
+import time
 
 comm = MPI.COMM_WORLD
 rank = comm.Get_rank()
-fv3core.set_backend("numpy")
-fv3core.set_rebuild(False)
-fv3core.set_validate_args(False)
-case_name = "c12_6ranks_baroclinic_dycore_microphysics"
-spec.set_namelist(case_name + "/input.nml")
-
-experiment_name = yaml.safe_load(open(case_name + "/input.yml", "r",))[
-    "experiment_name"
-]
-
-# set up of helper structures
-serializer = serialbox.Serializer(
-    serialbox.OpenModeKind.Read, case_name, "Generator_rank" + str(rank),
-)
-cube_comm = util.CubedSphereCommunicator(
-    comm, util.CubedSpherePartitioner(util.TilePartitioner(spec.namelist.layout)),
-)
 
-# get grid from serialized data
-grid_savepoint = serializer.get_savepoint("Grid-Info")[0]
-grid_data = {}
-grid_fields = serializer.fields_at_savepoint(grid_savepoint)
-for field in grid_fields:
-    grid_data[field] = serializer.read(field, grid_savepoint)
-    if len(grid_data[field].flatten()) == 1:
-        grid_data[field] = grid_data[field][0]
-grid = fv3core.testing.TranslateGrid(grid_data, rank).python_grid()
-spec.set_grid(grid)
-
-# set up domain decomposition
-layout = spec.namelist.layout
-partitioner = util.CubedSpherePartitioner(util.TilePartitioner(layout))
-communicator = util.CubedSphereCommunicator(comm, partitioner)
-
-# create a state from serialized data
-savepoint_in = serializer.get_savepoint("FVDynamics-In")[0]
-driver_object = fv3core.testing.TranslateFVDynamics([grid])
-input_data = driver_object.collect_input_data(serializer, savepoint_in)
-input_data["comm"] = communicator
-state = driver_object.state_from_inputs(input_data)
-
-# read in missing grid info for physics - this will be removed
-dwind = TranslateUpdateDWindsPhys(grid)
-missing_grid_info = dwind.collect_input_data(
-    serializer, serializer.get_savepoint("FVUpdatePhys-In")[0]
-)
+def print_for_rank0(msg: str):
+    '''Only prints when rank is 0. Flush immediately.''' 
+    if rank == 0:
+        print(f"[R{rank}]{msg}", flush=True)
 
-# initialize dynamical core and physics objects
-dycore = fv3core.DynamicalCore(
-    comm=communicator,
-    grid_data=spec.grid.grid_data,
-    grid_indexing=spec.grid.grid_indexing,
-    damping_coefficients=spec.grid.damping_coefficients,
-    config=spec.namelist.dynamical_core,
-    ak=state["atmosphere_hybrid_a_coordinate"],
-    bk=state["atmosphere_hybrid_b_coordinate"],
-    phis=state["surface_geopotential"],
+
+# Reuse infrastructure to read in grid variables
+# add path to integration test to reuse existing grid logic
+sys.path.append(
```

Comment:
I am going to pass on refactoring this, because it'll be gone soon and because I think this script will be use on `daint` mostly to get the data you want quickly (and on all arch)